game.py

```diff
@@ -1,41 +1,82 @@
-from random import randint
+from operator import itemgetter
+
+from random import choice, randint
 
 from exceptions import NimException
 
 
 class Bot():
+    subclasses = []
+
+    def __init_subclass__(cls, **kwargs):
+        super().__init_subclass__(**kwargs)
+        Bot.subclasses.append(cls)
+
+    def valid_pile_indexes(self, state):
```

Comment:
Thinking about it, an empty pile isn't invalid. So maybe we should find a better name for this method.